Engine/ac/string.cpp

```diff
@@ -49,6 +49,62 @@ int String_IsNullOrEmpty(const char *thisString)
     return 0;
 }
 
+const char * String_Join(void* arrobj, const char *separator)
+{
+    if(arrobj == nullptr)
+        return nullptr;
+
+    auto const& header = CCDynamicArray::GetHeader(arrobj);
+    const size_t count = header.ElemCount;
+    if(count == 0)
+        return CreateNewScriptString("");
+
+    if(separator == nullptr)
+        separator = "";
+
+    const uint32_t *handles;
+    handles= static_cast<const uint32_t*>(arrobj);
+
+    int sep_len, sep_ulen;
+    ustrlen2(separator, &sep_len, &sep_ulen);
+
+    std::vector<ScriptString*> script_strings{};
```

Comment:
This is not correct, the ScriptString is only a manager, but the registered objects are buffers of type `char*`.

Reply:
Ah, this is why the autocomplete was making no sense with the data.